systemtest: TestKeepUnsampled is flaky

https://apm-ci.elastic.co/blue/organizations/jenkins/apm-server%2Fapm-server-mbp/detail/PR-4243/1/pipeline

```
[2020-09-29T08:55:22.018Z] === RUN   TestKeepUnsampled/true
[2020-09-29T08:55:22.018Z]     sampling_test.go:61: 
[2020-09-29T08:55:22.018Z]         	Error Trace:	sampling_test.go:61
[2020-09-29T08:55:22.018Z]         	Error:      	"[{apm-8.0.0-transaction-000001 WggS2XQB8zK8KinSvzNi %!s(float64=0.2876821) map[@timestamp:2020-09-29T08:55:16.587Z agent:map[name:go version:0.0.0] ecs:map[version:1.6.0] event:map[ingested:2020-09-29T08:55:18.626171098Z outcome:unknown] host:map[architecture:i386 hostname:beowulf ip:127.0.0.1 name:beowulf os:map[platform:minix]] observer:map[ephemeral_id:421e2333-a73e-4430-a357-226671d6e405 hostname:apm-ci-immutable-ubuntu-1804-1601368514814223369 id:1c47a175-3900-4253-87c2-cd78369671b1 type:apm-server version:8.0.0 version_major:%!s(float64=8)] process:map[args:[/tmp/go-build445466795/b001/systemtest.test -test.testlogfile=/tmp/go-build445466795/b001/testlog.txt -test.timeout=10m0s -test.v=true] pid:%!s(float64=1) title:systemtest.test] processor:map[event:transaction name:transaction] service:map[language:map[name:go version:2.0] name:systemtest node:map[name:beowulf] runtime:map[name:gc version:2.0]] timestamp:map[us:%!s(float64=1.601369716587913e+15)] trace:map[id:b5da94c52eabd779b2de65019e12ddfa] transaction:map[duration:map[us:%!s(float64=0)] id:b5da94c52eabd779 name:sampled sampled:%!s(bool=true) span_count:map[dropped:%!s(float64=0) started:%!s(float64=0)] type:TestKeepUnsampled]] {"process":{"args":["/tmp/go-build445466795/b001/systemtest.test","-test.testlogfile=/tmp/go-build445466795/b001/testlog.txt","-test.timeout=10m0s","-test.v=true"],"pid":1,"title":"systemtest.test"},"agent":{"name":"go","version":"0.0.0"},"processor":{"name":"transaction","event":"transaction"},"observer":{"hostname":"apm-ci-immutable-ubuntu-1804-1601368514814223369","id":"1c47a175-3900-4253-87c2-cd78369671b1","type":"apm-server","ephemeral_id":"421e2333-a73e-4430-a357-226671d6e405","version":"8.0.0","version_major":8},"trace":{"id":"b5da94c52eabd779b2de65019e12ddfa"},"@timestamp":"2020-09-29T08:55:16.587Z","ecs":{"version":"1.6.0"},"service":{"node":{"name":"beowulf"},"name":"systemtest","runtime":{"name":"gc","version":"2.0"},"language":{"name":"go","version":"2.0"}},"host":{"hostname":"beowulf","os":{"platform":"minix"},"ip":"127.0.0.1","name":"beowulf","architecture":"i386"},"event":{"ingested":"2020-09-29T08:55:18.626171098Z","outcome":"unknown"},"transaction":{"duration":{"us":0},"name":"sampled","id":"b5da94c52eabd779","span_count":{"dropped":0,"started":0},"type":"TestKeepUnsampled","sampled":true},"timestamp":{"us":1601369716587913}}}]" should have 2 item(s), but has 1
[2020-09-29T08:55:22.018Z]         	Test:       	TestKeepUnsampled/true
[2020-09-29T08:55:22.018Z]     server.go:166: log file: /var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-4243/src/github.com/elastic/apm-server/systemtest/logs/TestKeepUnsampled_true/apm-server
[2020-09-29T08:55:22.018Z] --- FAIL: TestKeepUnsampled (4.90s)
[2020-09-29T08:55:22.018Z]     --- PASS: TestKeepUnsampled/false (2.43s)
[2020-09-29T08:55:22.018Z]     --- FAIL: TestKeepUnsampled/true (2.47s)
[2020-09-29T08:55:22.018Z] FAIL
[2020-09-29T08:55:22.018Z] FAIL	github.com/elastic/apm-server/systemtest	42.831s
```

--- a/systemtest/estest/search.go
+++ b/systemtest/estest/search.go
@@ -28,16 +28,20 @@ import (
 
 // ExpectDocs searches index with query, returning the results.
 //
-// If the search returns no results within 10 seconds (by default),
-// ExpectDocs will call t.Error().
+// ExpectDocs is equivalent to calling ExpectMinDocs with a minimum of 1.
 func (es *Client) ExpectDocs(t testing.TB, index string, query interface{}, opts ...RequestOption) SearchResult {
+	return es.ExpectMinDocs(t, 1, index, query, opts...)
+}
+
+// ExpectMinDocs searches index with query, returning the results.
+//
+// If the search returns fewer than min results within 10 seconds
+// (by default), ExpectMinDocs will call t.Error().
+func (es *Client) ExpectMinDocs(t testing.TB, min int, index string, query interface{}, opts ...RequestOption) SearchResult {
 	t.Helper()
 	var result SearchResult
-	_, err := es.Search(index).WithQuery(query).Do(
-		context.Background(), &result,
-		WithCondition(result.Hits.NonEmptyCondition()),
-	)
-	if err != nil {
+	opts = append(opts, WithCondition(result.Hits.NonEmptyCondition()))
+	if _, err := es.Search(index).WithQuery(query).Do(context.Background(), &result, opts...); err != nil {
 		t.Error(err)
 	}
 	return result
@@ -77,7 +81,13 @@ type SearchHits struct {
 
 // NonEmptyCondition returns a ConditionFunc which will return true if h.Hits is non-empty.
 func (h *SearchHits) NonEmptyCondition() ConditionFunc {
-	return func(*esapi.Response) bool { return len(h.Hits) != 0 }
+	return h.MinHitsCondition(1)
+}
+
+// MinHitsCondition returns a ConditionFunc which will return true if the number of h.Hits
+// is at least min.
+func (h *SearchHits) MinHitsCondition(min int) ConditionFunc {
+	return func(*esapi.Response) bool { return len(h.Hits) >= min }
 }
 
 type SearchHit struct {

--- a/systemtest/sampling_test.go
+++ b/systemtest/sampling_test.go
@@ -49,15 +49,15 @@ func TestKeepUnsampled(t *testing.T) {
 			tracer.StartTransaction("unsampled", transactionType).End()
 			tracer.Flush(nil)
 
-			result := systemtest.Elasticsearch.ExpectDocs(t, "apm-*", estest.TermQuery{
-				Field: "transaction.type",
-				Value: transactionType,
-			})
-
 			expectedTransactionDocs := 1
 			if keepUnsampled {
 				expectedTransactionDocs++
 			}
+
+			result := systemtest.Elasticsearch.ExpectMinDocs(t, expectedTransactionDocs, "apm-*", estest.TermQuery{
+				Field: "transaction.type",
+				Value: transactionType,
+			})
 			assert.Len(t, result.Hits.Hits, expectedTransactionDocs)
 		})
 	}

